server/files/configure_misp.sh

```diff
@@ -101,26 +103,27 @@ apply_updates() {
 init_user() {
     # Create the main user if it is not there already
     sudo -u www-data /var/www/MISP/app/Console/cake userInit -q 2>&1 > /dev/null
-    echo "... setting admin email to '${ADMIN_EMAIL}'"
-    sudo -u www-data /var/www/MISP/app/Console/cake Admin setSetting -q "MISP.email" ${ADMIN_EMAIL}
+
     echo "UPDATE misp.users SET email = \"${ADMIN_EMAIL}\" WHERE id = 1;" | ${MYSQLCMD}
+
     if [ ! -z "$ADMIN_ORG" ]; then
         echo "UPDATE misp.organisations SET name = \"${ADMIN_ORG}\" where id = 1;" | ${MYSQLCMD}
     fi
 
-    if [ "$AUTOCONF_ADMIN_KEY" == "true" ]; then
-        if [ ! -z "$ADMIN_KEY" ]; then
-            echo "... setting admin key to '${ADMIN_KEY}'"
-            CHANGE_CMD=(sudo -u www-data /var/www/MISP/app/Console/cake User change_authkey 1 "${ADMIN_KEY}")
-        else
-            echo "... regenerating admin key (set \$ADMIN_KEY if you want it to change)"
-            CHANGE_CMD=(sudo -u www-data /var/www/MISP/app/Console/cake User change_authkey 1)
-        fi
-        ADMIN_KEY=`${CHANGE_CMD[@]} | awk 'END {print $NF; exit}'`
-        echo "... admin user key set to '${ADMIN_KEY}'"
+    if [ -n "$ADMIN_KEY" ]; then
+        echo "... setting admin key to '${ADMIN_KEY}'"
+        CHANGE_CMD=(sudo -u www-data /var/www/MISP/app/Console/cake User change_authkey 1 "${ADMIN_KEY}")
+    elif [ -z "$ADMIN_KEY" ] && [ "$AUTOGEN_ADMIN_KEY" == "true" ]; then
+        echo "... regenerating admin key (set \$ADMIN_KEY if you want it to change)"
+        CHANGE_CMD=(sudo -u www-data /var/www/MISP/app/Console/cake User change_authkey 1)
     else
         ADMIN_KEY=""
-        echo "... admin user key auto configuration disabled"
+        echo "... admin user key auto generation disabled"
+    fi
+
+    if [[ -v CHANGE_CMD[@] ]]; then
```

Comment:
If we get here, isn't redundant setting ADMIN_KEY to an empty string

Reply:
Possibly. As you can see, it was just leftover from the old branch.

Reply:
If that is the case, shall we remove it from line 120?

Reply:
I think that would be safe. Will do.